vector/src/main/java/im/vector/app/features/session/VectorSessionStore.kt

```diff
@@ -14,48 +14,46 @@
  * limitations under the License.
  */
 
-package im.vector.app.features.onboarding.store
+package im.vector.app.features.session
 
 import android.content.Context
 import androidx.datastore.core.DataStore
 import androidx.datastore.preferences.core.Preferences
 import androidx.datastore.preferences.core.edit
 import androidx.datastore.preferences.core.stringPreferencesKey
 import androidx.datastore.preferences.preferencesDataStore
-import im.vector.app.core.extensions.removeKeysWithPrefix
 import im.vector.app.features.onboarding.FtueUseCase
 import kotlinx.coroutines.flow.first
 
-private val Context.dataStore: DataStore<Preferences> by preferencesDataStore(name = "vector_onboarding")
-
 /**
  * Local storage for:
  * - messaging use case (Enum/String)
  */
-class OnboardingStore constructor(
+class VectorSessionStore constructor(
         private val context: Context,
-        private val myUserId: String
+        myUserId: String
 ) {
 
+    private val Context.dataStore: DataStore<Preferences> by preferencesDataStore(name = "vector_session_store_$myUserId")
```

Comment:
I am wondering if directly using the `userId` may lead to some trouble: special chars - low risk, length - higher risk. Maybe using a hash of it would be better.

Reply:
very good point, I keep forgetting that user-id is actually a matrix id rather than a uuid, will change

Reply:
Yes. "Matrix Id" is not really a specific concept, but it is sometimes used for user ids. https://spec.matrix.org/v1.1/appendices/#identifier-grammar is an interesting reading on the subject

Reply:
https://github.com/vector-im/element-android/pull/5009/commits/cbdeb54fdbd687561f043ad807d0620c8a722523